TLM/TLM/Util/Extensions/VehicleExtensions.cs

```diff
@@ -36,20 +36,23 @@ public static class VehicleExtensions {
             vehicle.m_flags.IsFlagSet(Vehicle.Flags.WaitingPath);
 
         /// <summary>Determines the <see cref="ExtVehicleType"/> for a vehicle.</summary>
-        /// <param name="vehicle">The vehocle to inspect.</param>
+        /// <param name="vehicle">The vehicle to inspect.</param>
         /// <returns>The extended vehicle type.</returns>
-        public static ExtVehicleType ToExtVehicleType(this ref Vehicle vehicle) {
-            var vehicleId = vehicle.Info.m_instanceID.Vehicle;
-            var vehicleAI = vehicle.Info.m_vehicleAI;
-            var emergency = vehicle.m_flags.IsFlagSet(Vehicle.Flags.Emergency2);
+        public static ExtVehicleType ToExtVehicleType(this ref Vehicle vehicle)
+            => vehicle.IsValid()
+                ? vehicle.Info.m_instanceID.Vehicle.ToExtVehicleType()
```

Comment:
BTW, It won't work at all. You can't access the `Vehicle` ID via `m_instanceID`, that field is not for what you expect and is always empty. It's for prefab pooling, used only for buildings and citizens(some code requires empty prefab to work and creating prefab from scratch takes a ton of time + generates garbage).

Reply:
How to access the id from the `Vehicle` in that case?

Reply:
Impossible, you need to pass it, always

Reply:
reverted to original approach that inspects vehicle ai

Reply:
in C/C++ having a pointer to array element and pointer to array start, you could find pointer difference (divided by element size). Can this trick be done with ref to vehicle and ref to vehicle array start?